dill/_dill.py

```diff
@@ -317,6 +316,26 @@ class PicklingWarning(PickleWarning, PicklingError):
 class UnpicklingWarning(PickleWarning, UnpicklingError):
     pass
 
+def _getopt(settings, key, arg=None, *, kwds=None):
+    """Get option from named argument 'arg' or 'kwds', falling back to settings.
+
+    Examples:
+
+        # With an explicitly named argument:
+        protocol = int(_getopt(settings, 'protocol', protocol))
+
+        # With a named argument in **kwds:
+        self._byref = _getopt(settings, 'byref', kwds=kwds)
+    """
+    # Sanity check, it's a bug in calling code if False.
+    assert kwds is None or arg is None
+    if kwds is not None:
+        arg = kwds.pop(key, None)
+    if arg is not None:
+        return arg
+    else:
+        return settings[key]
```

Comment:
It's a bit unorthodox that the alternate is given as the first two values, and then the arg/kwd you are primarily interested in is given after that.  Consider the syntax of `getattr` and similar.

So, this function pops `key` out of `kwds` (isn't get a better choice?), and if `key` is found and not `None`, then it is returned.  If not found, then use `arg` if `arg` is provided, unless it's `None`... then default back to `settings[key]`.  It's also expected that `arg` and `kwds` are never both given... so essentially, you have two one-liners in a single function, where any interaction between the two one-liners is an error.  I'm not seeing the utility of this function, in that case.  I could see using it, maybe, if you wanted to hide the import of `settings` within the function -- so `_getopt(key, arg=None, *, kwds=None)`.